File: eMovieFinder/eMovieFinder.Model/SearchObjects/MovieActorSearchObject.cs
namespace eMovieFinder.Model.SearchObjects
{
    public class MovieActorSearchObject : BaseSearchObject
    {
        public string? FullName { get; set; }
        public int? MovieId { get; set; }
        public bool? IsMovieIncluded { get; set; }
        public bool? IsActorIncluded { get; set; }
    }
}

File: eMovieFinder-UI/emovie_finder_desktop/lib/src/models/searchobjects/movie_actor_search_object.dart
import 'package:json_annotation/json_annotation.dart';
import 'base_search_object.dart';
part 'movie_actor_search_object.g.dart';

@JsonSerializable()
class MovieActorSearchObject extends BaseSearchObject {
  final bool? isMovieIncluded;
  final bool? isActorIncluded;

  MovieActorSearchObject({
    super.pageNumber,
    super.pageSize,
    super.orderBy,
    super.isDescending,
    this.isMovieIncluded,
    this.isActorIncluded
  });

  factory MovieActorSearchObject.fromJson(Map<String, dynamic> json) => _$MovieActorSearchObjectFromJson(json);
  Map<String, dynamic> toJson() => _$MovieActorSearchObjectToJson(this);
}